onnxruntime/test/qnn_ctx_gen/command_args_parser.cc

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) 2023 NVIDIA Corporation.
+// Licensed under the MIT License.
+
+#include "command_args_parser.h"
+
+#include <string.h>
+#include <iostream>
+#include <sstream>
+#include <string_view>
+#include <unordered_map>
+
+// Windows Specific
+#ifdef _WIN32
+#include "getopt.h"
+#include "windows.h"
+#else
+#include <unistd.h>
+#endif
+
+#include <core/graph/constants.h>
+#include <core/platform/path_lib.h>
+#include <core/optimizer/graph_transformer_level.h>
+
+#include "test_configuration.h"
+
+namespace onnxruntime {
+namespace qnnctxgen {
+
+/*static*/ void CommandLineParser::ShowUsage() {
+  printf(
+      "onnxruntime_qnn_ctx_gen [options...] model1_path,model2_path\n"
+      "Options:\n"
+      "\t-v: Show verbose information.\n"
+      "\t-C: Specify session configuration entries as key-value pairs: -C \"<key1>|<value1> <key2>|<value2>\" \n"
+      "\t    Refer to onnxruntime_session_options_config_keys.h for valid keys and values. \n"
+      "\t    [Example] -C \"ep.context_enable|1\" \n"
+      "\t-i: Specify QNN EP specific runtime options as key value pairs. Different runtime options available are: \n"
+      "\t    [Usage]: -i '<key1>|<value1> <key2>|<value2>'\n"
+      "\n"
+      "\t    [backend_path]: QNN backend path. e.g '/folderpath/libQnnHtp.so', '/winfolderpath/QnnHtp.dll'. default to HTP backend\n"
+      "\t    [vtcm_mb]: QNN VTCM size in MB. default to 0(not set).\n"
+      "\t    [htp_graph_finalization_optimization_mode]: QNN graph finalization optimization mode, options: '0', '1', '2', '3', default is '0'.\n"
+      "\t    [soc_model]: The SoC Model number. Refer to QNN SDK documentation for specific values. Defaults to '0' (unknown). \n"
+      "\t    [htp_arch]: The minimum HTP architecture. The driver will use ops compatible with this architecture. eg: '0', '68', '69', '73', '75'. Defaults to '0' (none). \n"
+      "\t    [enable_htp_fp16_precision]: Enable the HTP_FP16 precision so that the float32 model will be inferenced with fp16 precision. \n"
+      "\t    Otherwise, it will be fp32 precision. Only works for float32 model. Defaults to '0' (with FP32 precision.). \n"
+      "\t    [enable_htp_weight_sharing]: Allows common weights across graphs to be shared and stored in a single context binary. Defaults to '1' (enabled).\n"
```

Comment:
can enable_htp_weight_sharing be set internally depending on if the user passing in multiple models or not?

Reply:
I think it's still useful to provide this option. for example set it to 0 and compare the result the with 1. I personally set it to 0 so that I can debug the workflow on Windows before QNN 2.26.